pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -213,6 +256,8 @@ func (s *IPBlocksInfoService) createOrUpdateIPBlocksInfo(ctx context.Context, ip
 	if incremental {
 		ipBlocksInfo.ExternalIPCIDRs = s.mergeIPCidrs(ipBlocksInfoOld.ExternalIPCIDRs, ipBlocksInfo.ExternalIPCIDRs)
 		ipBlocksInfo.PrivateTGWIPCIDRs = s.mergeIPCidrs(ipBlocksInfoOld.PrivateTGWIPCIDRs, ipBlocksInfo.PrivateTGWIPCIDRs)
+		ipBlocksInfo.ExternalIPRanges = s.mergeIPRanges(ipBlocksInfoOld.ExternalIPRanges, ipBlocksInfo.ExternalIPRanges)
+		ipBlocksInfo.PrivateTGWIPRanges = s.mergeIPRanges(ipBlocksInfoOld.PrivateTGWIPRanges, ipBlocksInfo.PrivateTGWIPRanges)
```

Comment:
Just to confirm if MergeArraysWithoutDuplicate is enough or we need mergeIPRanges as we do not have subnet allocated from ip ranges?

Reply:
I did test and NSX has overlap checking while adding the ipblocks to the connectivity profile. so MergeArraysWithoutDuplicate should be enough